### PR TITLE
Added Indexes and Fixed Database

### PR DESCRIPTION
**Pull request by EdgardoJ-Gonzalez**
Updated Association Table
Added Indexes
Added sqlalchemy to requirements

### CODE REVIEW

File: src/openbook/models.py [hunk 1]
```diff
@@ -1,57 +1,94 @@
...
+    COMPLETED = "completed"
+
+
+class UserBook(Base):
+    """Represents the association between users and books."""
+
+    __tablename__ = "user_book"
+
+    # Association Table Columns
+    book_id = Column(Integer, ForeignKey("books.id"), primary_key=True)
```
Comment (jonathan-d-zhang) [inline src/openbook/models.py:23]: We should use the declarative form when possible: reduces boilerplate and adds type hints.

https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#declarative-table-with-mapped-column

File: src/openbook/models.py [hunk 2]
```diff
@@ -1,57 +1,94 @@
-from sqlalchemy import Column, String, Integer, ForeignKey
+import enum
+
+from sqlalchemy import Column, Enum, ForeignKey, Index, Integer, String
 from sqlalchemy.orm import declarative_base, relationship
 
 Base = declarative_base()
```
Comment (jonathan-d-zhang) [inline src/openbook/models.py:6]:
```suggestion
from sqlalchemy.orm import MappedAsDataclass, DeclarativeBase
class Base(MappedAsDataclass, DeclarativeBase, kw_only=True):
    pass
```

File: src/openbook/models.py [hunk 3]
```diff
@@ -1,57 +1,94 @@
...
+
+
+class UserBook(Base):
+    """Represents the association between users and books."""
+
+    __tablename__ = "user_book"
+
+    # Association Table Columns
+    book_id = Column(Integer, ForeignKey("books.id"), primary_key=True)
+    user_id = Column(Integer, ForeignKey("users.id"), primary_key=True)
```
Comment (jonathan-d-zhang) [inline src/openbook/models.py:24]: Are we sure that having two columns with `primary_key=True` creates a composite primary key? I also see that you have the `__table_args__` below... I don't think we need the kwarg and the table arg.

File: src/openbook/models.py [hunk 4]
```diff
@@ -1,57 +1,94 @@
...
+
+    RECOMMENDED = "recommended"
+    READING = "reading"
+    COMPLETED = "completed"
+
+
+class UserBook(Base):
+    """Represents the association between users and books."""
+
+    __tablename__ = "user_book"
```
Comment (jonathan-d-zhang) [inline src/openbook/models.py:20]: Be consistent in the table name: either all plural (`users_books`), or don't make the user table plural (`user`). Same with `books`.

File: src/openbook/models.py [hunk 5]
```diff
@@ -1,57 +1,94 @@
...
 
+    # Table Columns
     id = Column(Integer, primary_key=True, autoincrement=True)
     email = Column(String, unique=True, nullable=False)
     name = Column(String)
-    password = Column(String)
 
-    completed_list = relationship("Completed_list", back_populates= "users", uselist=False)
+    # Many-to-many relationship with Book through UserBook
+    user_book_entries = relationship("UserBook", back_populates="user")
```
Comment (jonathan-d-zhang) [inline src/openbook/models.py:59]:
```suggestion
    books = relationship("UserBook", back_populates="user")
```

File: src/openbook/models.py [hunk 6]
```diff
@@ -1,57 +1,94 @@
...
+    __tablename__ = "books"
 
+    # Table Columns
     id = Column(Integer, primary_key=True, autoincrement=True)
     title = Column(String, nullable=False)
-    author = Column(String, nullable=False)
 
-    reading_list_book = relationship("Reading_list_book", back_populates="books")
+    # Many-to-many relationship with User through UserBook
+    user_book_entries = relationship("UserBook", back_populates="book")
```
Comment (jonathan-d-zhang) [inline src/openbook/models.py:74]:
```suggestion
    users = relationship("UserBook", back_populates="book")
```

**Review by jonathan-d-zhang — CHANGES_REQUESTED** [review]
Schema looks good overall, just some minor changes to align with best practices.

**Review by jonathan-d-zhang — DISMISSED** [review]
Minor nits. Looks good though

File: src/openbook/models.py [hunk 7]
```diff
@@ -19,76 +21,72 @@ class UserBook(Base):
...
 
-    # Many-to-many relationship with User through UserBook
-    user_book_entries = relationship("UserBook", back_populates="book")
+    # Many-to-many relationship with Author through AuthorBook with type hints
+    author: Mapped[list["AuthorBook"]] = relationship("AuthorBook", back_populates="book")
 
-    # Many-to-many relationship with Author through AuthorBook
-    authors = relationship("AuthorBook", back_populates="book")
+    # Table Columns with type hints
+    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
```
[the remark this answers is not in the document]
Comment (EdgardoJ-Gonzalez) [reply src/openbook/models.py:74]: Do you want me to remove this from all of them due to the type hints?

File: src/openbook/models.py [hunk 8]
```diff
@@ -19,76 +21,72 @@ class UserBook(Base):
...
 
-    # Table Columns
-    id = Column(Integer, primary_key=True, autoincrement=True)
-    title = Column(String, nullable=False)
+    # Many-to-many relationship with User through UserBook with type hints
+    user: Mapped[list["UserBook"]] = relationship("UserBook", back_populates="book")
 
-    # Many-to-many relationship with User through UserBook
-    user_book_entries = relationship("UserBook", back_populates="book")
+    # Many-to-many relationship with Author through AuthorBook with type hints
```
[the remark this answers is not in the document]
Comment (EdgardoJ-Gonzalez) [reply src/openbook/models.py:70]: Why remove the comments?

[the remark this answers is not in the document]
Comment (jonathan-d-zhang) [reply src/openbook/models.py:70]: Redundant

File: src/openbook/models.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (jonathan-d-zhang) [reply src/openbook/models.py:74]: That would be appreciated.

**Review by jonathan-d-zhang — APPROVED** [review]
:shipit:

**Pull request merged**